main.py

```diff
@@ -77,28 +78,23 @@ def root():
 
 
 
-@app.route('/adopt/<listing_id>', methods=['GET', 'POST'])
+@app.route('/adopt/<listing_id>', methods=['POST'])
 def adopt(listing_id):
-    form = AdoptionForm()
-        
     listing = get_listing(listing_id)
-    if not listing:
-        app.logger.error(f"Failed to get listing: {listing_id}")
-        abort(404, description=f"Failed to get listing: {listing_id}")
-
-    if form.validate_on_submit():
-        adopter_name = form.name.data
-        adopter_email = form.email.data
-        email_message = form.message.data
-        
-        poster_email = listing.user_email
+    # if not listing:
```

Comment:
Would be good to add error check if listing cannot be found

Reply:
Added